subprojects/groovy-nio/src/main/java/org/apache/groovy/nio/extensions/NioExtensions.java

```diff
@@ -101,6 +101,36 @@ public static long size(Path self) throws IOException {
         return Files.size(self);
     }
 
+    /**
+     * Provide the extension of this path.
+     *
+     * @param self a path object
+     * @return the extension of this path (not including the dot), or an empty string if it doesn't have one.
+     * @since 5.0.0
+     */
+    public static String getExtension(Path self) {
+        final Path fileName = self.getFileName();
+        final String name = fileName == null ? "" : fileName.toString();
```

Comment:
If we can resue this part, add a `Path.name()` like [`Path.name` in Kotlin](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io.path/java.nio.file.-path/name.html)?

Reply:
You want `Path.getName()`, with shorthand `Path.name` as a shorthand for `Path.getFileName()`/`Path.fileName`?
That could work but there is the existing `getName(int index)` method which might confuse some folks.

Reply:
Yeah, that's what I'm afraid.